crates/sui-graphql-rpc/src/server/version.rs

```diff
@@ -108,12 +109,18 @@ pub(crate) async fn check_version_middleware<B>(
 
 /// Mark every outgoing response with a header indicating the precise version of the RPC that was
 /// used (including the patch version).
-pub(crate) async fn set_version_middleware<B>(request: Request<B>, next: Next<B>) -> Response {
+pub(crate) async fn set_version_middleware<B>(
+    State(version): State<Version>,
+    request: Request<B>,
+    next: Next<B>,
+) -> Response {
     let mut response = next.run(request).await;
     let headers = response.headers_mut();
+    let version = version.to_string();
     headers.insert(
         VERSION_HEADER.clone(),
-        HeaderValue::from_static(RPC_VERSION_FULL),
+        HeaderValue::from_str(version.as_str())
+            .unwrap_or_else(|_| HeaderValue::from_static(env!("CARGO_PKG_VERSION"))),
```

Comment:
No need for this unwrap_or_else -- if the configured version doesn't work, we should just fail. We should find out pretty soon that something's messed up because we set this on every request, so the initial health check will catch it.

Reply:
Now that we have the `full` field, I can just pass that.